Engine/RealTime/BaseRealTimeHandler.cs

```diff
@@ -101,16 +100,30 @@ protected void Setup(DateTime start, DateTime end, Language language, DateTime?
                 {
                     _implementsOnEndOfDaySymbol = true;
                 }
+
+                // Also determine if we are using the soon to be deprecated EOD so we don't use it
+                // unnecessarily and post messages about its deprecation to the user
+                var eodMethod = Algorithm.GetType().GetMethod("OnEndOfDay", Type.EmptyTypes);
+                if (eodMethod != null && eodMethod.DeclaringType != typeof(QCAlgorithm))
+                {
+                    _implementsOnEndOfDay = true;
+                }
             }
             else if (language == Language.Python)
             {
                 var wrapper = Algorithm as AlgorithmPythonWrapper;
                 _implementsOnEndOfDaySymbol = wrapper.IsOnEndOfDayImplemented;
+
+                // Since we can't differentiate the methods in Python we have to set this to the same
+                _implementsOnEndOfDay = _implementsOnEndOfDaySymbol;
```

Comment:
When we get the python method using `GetPythonMethod` isn't there some property of it we could use to determine this? for example the argument count

Reply:
I found a way around this, we can use python lib inspect which is included in the python install to get the methods sig and get a count on the parameters 👍🏽